### PR TITLE
Fix: Add convert-custom-slash-for-codex command (Closes #247)

### LINKED ISSUE #247: Add convert-custom-slash-for-codex command to convert Claude commands to Codex prompts
## Description

Add a new Claude custom slash command `convert-custom-slash-for-codex` that converts existing Claude custom slash commands (from `.claude/commands/`) into Codex-optimized prompt files (for `.codex/prompts/`).

This command will enable seamless conversion of Claude commands to Codex format, allowing developers to use the same commands in both AI assistants with appropriate format optimizations.

## Context

This repository uses both Claude (via `.claude/commands/`) and Codex (via `~/.codex/prompts/`). Currently, there's a basic Makefile target (`create-codex-custom-prompts`) that only copies files without format conversion. Codex prompts require:

1. **YAML frontmatter** with `description:` and `argument-hint:` fields
2. **Placeholder syntax** (`$1`, `$2`, `$ARGUMENTS`, `$NAMED_PARAM`) instead of markdown parameter references
3. **Literal dollar signs** using `$$` syntax

The conversion command should intelligently transform Claude command structure to Codex format while preserving functionality and improving discoverability.

## Acceptance Criteria

- [ ] Command accepts a Claude command name as parameter (e.g., `fix-issue`, `fix-linter`)
- [ ] Reads source command from `.claude/commands/{name}.md`
- [ ] Extracts description from command content (title or first paragraph)
- [ ] Generates appropriate YAML frontmatter with `description:` and `argument-hint:`
- [ ] Converts parameter references to Codex placeholder syntax (`$1`, `$2`, `$NAMED_PARAM`)
- [ ] Handles literal dollar signs by converting to `$$`
- [ ] Saves converted prompt to `~/.codex/prompts/{name}.md`
- [ ] Creates `~/.codex/prompts/` directory if it doesn't exist
- [ ] Provides clear output showing conversion status
- [ ] Handles errors gracefully (missing source file, invalid format, etc.)

### Architecture
- **Layer**: Interface adapters (CLI command)
- **Location**: `.claude/commands/convert-custom-slash-for-codex.md`
- **Dependencies**: None (uses standard file I/O and string manipulation)

#### Command Structure
- **Command name**: `convert-custom-slash-for-codex`
- **Parameter**: `COMMAND_NAME` (required) - name of Claude command to convert (without `.md` extension)
- **Usage**: `/convert-custom-slash-for-codex COMMAND_NAME=fix-issue`

#### Conversion Logic

1. **Read Source Command**:
   - Read from `.claude/commands/{COMMAND_NAME}.md`
   - Validate file exists

2. **Extract Metadata**:
   - **Description**: Extract from first heading or first paragraph (first 100-150 characters)
   - **Argument hints**: Analyze command content for parameter patterns:
     - Issue numbers: `#123` → `ISSUE_NUMBER=#123`
     - File paths: `FILE=path/to/file`
     - Named parameters in command text

3. **Transform Content**:
   - **Add YAML frontmatter**:
     ```yaml
     ---
     description: [extracted description]
     argument-hint: [COMMAND_NAME=<name>] [ISSUE_NUMBER=#123] [FILE=<path>]
     ---
     ```
   - **Convert parameter references**:
     - `#{issue_number}` → `$ISSUE_NUMBER` or `$1` (if positional)
     - `{file_path}` → `$FILE` or `$2` (if positional)
     - Multiple parameters → `$1`, `$2`, `$3`, etc.
   - **Handle dollar signs**: `$` → `$$` (except in placeholders)
   - **Preserve markdown structure**: Keep headings, code blocks, lists, etc.

4. **Save Output**:
   - Create `~/.codex/prompts/` if missing
   - Write to `~/.codex/prompts/{COMMAND_NAME}.md`
   - Preserve file permissions

#### Example Conversion

**Input** (`.claude/commands/fix-issue.md`):
```markdown
# Fix Issue #{issue_number}

You are tasked with fixing GitHub issue #{issue_number}...
```

**Output** (`~/.codex/prompts/fix-issue.md`):
```markdown
---
description: Fix a specified GitHub issue by analyzing, implementing, testing, and creating a PR
argument-hint: [ISSUE_NUMBER=#123]
---
# Fix Issue $ISSUE_NUMBER

You are tasked with fixing GitHub issue $ISSUE_NUMBER...
```

### Files to Create/Modify

- **Create**: `.claude/commands/convert-custom-slash-for-codex.md`
- **Update**: `.claude/commands/README.md` (add entry for new command)

### Constraints
- Must preserve all command functionality during conversion
- Must handle edge cases (missing files, malformed content, special characters)
- Must not modify source Claude commands (read-only)
- Must work with all existing commands in `.claude/commands/`

### Unit Tests
- Test description extraction from various formats
- Test parameter reference conversion
- Test dollar sign escaping
- Test YAML frontmatter generation
- Test error handling (missing file, invalid format)

### Manual Testing
- Convert each existing command:
  - `fix-issue` → should handle `ISSUE_NUMBER` parameter
  - `fix-linter` → should handle no parameters or optional parameters
  - `fix-pr-review` → should handle `PR_NUMBER` parameter
  - `create-github-issue` → should handle multiple optional parameters
- Verify converted prompts work in Codex
- Test with commands that have complex parameter patterns

### Integration Testing
- Verify converted prompts appear in Codex slash command menu
- Test placeholder expansion in Codex
- Verify argument hints display correctly

## Related Context

- **Related to**: Codex custom prompts documentation: https://developers.openai.com/codex/custom-prompts
- **Related to**: Existing Makefile target `create-codex-custom-prompts` (basic copy, needs enhancement)
- **Documentation**: 
  - `.claude/commands/README.md` - Current custom commands structure
  - `.codex/prompts/README.md` - Codex prompts documentation (needs update)
  - `docs/ai-agents/agent-skills.md` - Agent Skills vs Custom Commands comparison

### Codex Prompt Format Requirements

Based on [Codex documentation](https://developers.openai.com/codex/custom-prompts):

1. **YAML Frontmatter**:
   - `description:` - Shown in slash command menu
   - `argument-hint:` - Documents expected parameters (optional)

2. **Placeholders**:
   - Positional: `$1`, `$2`, ..., `$9`
   - Named: `$NAMED_PARAM` (uppercase)
   - All arguments: `$ARGUMENTS`
   - Literal `$`: `$$`

3. **File Location**:
   - Must be in `~/.codex/prompts/` (user's home directory)
   - Codex scans only top-level Markdown files

### Conversion Strategy

1. **Simple commands** (no parameters): Direct conversion with description extraction
2. **Commands with issue/PR numbers**: Convert to `$ISSUE_NUMBER` or `$PR_NUMBER`
3. **Commands with file paths**: Convert to `$FILE` or `$FILES`
4. **Commands with multiple parameters**: Use positional (`$1`, `$2`) or named parameters

### Future Enhancements

- Batch conversion: Convert all commands at once
- Validation: Verify converted prompts are valid Codex format
- Reverse conversion: Convert Codex prompts back to Claude format
- Interactive mode: Prompt for parameter names and descriptions

### PR DESCRIPTION
**Pull request by hiromaily**
## Description

This PR implements a new Claude custom slash command that converts existing Claude commands to Codex-optimized prompt format, addressing issue #247.

The command enables seamless conversion from `.claude/commands/` to `~/.codex/prompts/` with appropriate format transformations for Codex compatibility.

### New Files
- **`.claude/commands/convert-custom-slash-for-codex.md`**: Main command implementation
  - Comprehensive conversion process with 7 detailed steps
  - Metadata extraction (description, argument hints)
  - Parameter placeholder conversion (e.g., `#{issue_number}` → `$ISSUE_NUMBER`)
  - Dollar sign escaping for literal `$` symbols
  - YAML frontmatter generation
  - Error handling for common scenarios
  - Complete example conversions

### Modified Files
- **`.claude/commands/README.md`**: Added documentation for new command
  - New section describing the command and its features
  - Usage example in the Usage section

### Conversion Logic
1. **Metadata Extraction**:
   - Description from first heading (100-150 char limit)
   - Argument hints from parameter patterns analysis

2. **Content Transformation**:
   - YAML frontmatter with `description:` and `argument-hint:` fields
   - Parameter conversion: `#{param}` or `{param}` → `$PARAM_NAME`
   - Dollar sign escaping: `$` → `$$` (except placeholders)
   - Markdown structure preservation

3. **Error Handling**:
   - Missing parameter validation
   - Source file existence check
   - Directory creation errors
   - Write permission errors

### Example Conversion

**Input** (`.claude/commands/fix-issue.md`):
```markdown
# Fix Issue #{issue_number}

You are tasked with fixing GitHub issue #{issue_number}...
```

**Output** (`~/.codex/prompts/fix-issue.md`):
```markdown
---
description: Fix Issue $ISSUE_NUMBER
argument-hint: ISSUE_NUMBER=#123
---
# Fix Issue $ISSUE_NUMBER

You are tasked with fixing GitHub issue $ISSUE_NUMBER...
```

### Manual Testing Plan
The command can be tested by:
1. Running `/convert-custom-slash-for-codex COMMAND_NAME=fix-issue`
2. Verifying output file: `cat ~/.codex/prompts/fix-issue.md`
3. Checking YAML frontmatter validity
4. Confirming parameter conversions
5. Testing with all existing commands:
   - `fix-issue` → `$ISSUE_NUMBER` parameter
   - `fix-linter` → No parameters
   - `fix-pr-review` → `$PR_NUMBER` parameter
   - `create-github-issue` → Multiple optional parameters

### Verification Status
- [x] Command structure follows existing command patterns
- [x] Comprehensive documentation with examples
- [x] Error handling covers common scenarios
- [x] Conversion rules well-defined
- [x] YAML frontmatter format specified
- [x] README.md updated with command entry
- [x] `make go-lint` passes
- [x] `make tidy` passes
- [x] `make check-build` passes

### Design Decisions
- **Prompt-based command**: This is an AI instruction file (not a programmatic tool), which means the conversion happens when invoked by Claude
- **Read-only source**: Never modifies `.claude/commands/` files
- **Overwrite behavior**: Overwrites existing files in `~/.codex/prompts/` without warning
- **Parameter detection**: Pattern-based detection using common patterns (`#{param}`, `{param}`)

### Limitations
- Does not validate YAML frontmatter syntax
- Does not test converted prompts in Codex automatically
- Manual review of converted prompts recommended
- Pattern-based parameter detection may miss unusual formats

## Related Context

- **Issue**: #247
- **Related Documentation**: 
  - [Codex Custom Prompts](https://developers.openai.com/codex/custom-prompts)
  - `.claude/commands/README.md` - Command structure
  - `docs/ai-agents/agent-skills.md` - Agent Skills comparison

## Additional Notes

This command complements the existing Makefile target `create-codex-custom-prompts` but provides intelligent conversion with format optimization rather than simple file copying.

Future enhancements could include:
- Batch conversion mode (convert all commands at once)
- Validation of converted Codex format
- Reverse conversion (Codex → Claude)
- Interactive parameter name selection

Closes #247

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/commands/README.md [hunk 1]
```diff
@@ -76,6 +87,7 @@ In Claude Desktop, you can use these commands as slash commands. For example:
 - `/fix-issue #123` - Fix issue #123
 - `/fix-linter` - Fix linter errors
 - `/fix-pr-review #456` - Address review comments for PR #456
+- `/convert-custom-slash-for-codex COMMAND_NAME=fix-issue` - Convert fix-issue command to Codex format
```
Comment (hiromaily) [inline .claude/commands/README.md]: like command `/fix-issue`, `COMMAAND_NAME` need to be omitted to run this custom slash command
